core/src/main/java/org/testcontainers/utility/ImageNameSubstitutor.java

```diff
@@ -0,0 +1,130 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.UnstableAPI;
+
+import java.util.function.Function;
+
+/**
+ * An image name substitutor converts a Docker image name, as may be specified in code, to an alternative name.
+ * This is intended to provide a way to override image names, for example to enforce pulling of images from a private
+ * registry.
+ * <p>
+ * This is marked as @{@link UnstableAPI} as this API is new. While we do not think major changes will be required, we
+ * will react to feedback if necessary.
+ */
+@Slf4j
+@UnstableAPI
+public abstract class ImageNameSubstitutor implements Function<DockerImageName, DockerImageName> {
+
+    @VisibleForTesting
+    static ImageNameSubstitutor instance;
+
+    @VisibleForTesting
+    static ImageNameSubstitutor defaultImplementation = new DefaultImageNameSubstitutor();
```

Comment:
WDYT about using `ConfigurationFileImageNameSubstitutor` directly, since we no longer expose `DefaultImageNameSubstitutor` to public?

Reply:
Ah, there's a reason for this: `DefaultImageNameSubstitutor` will be the place where we hook in the 'prefixing' image name substitutor in 1.15.1. 

Right now the default substitutor is pretty useless, but I'd prefer not to remove it given that I think we'll have to re-add it very soon.

Reply:
ah, okay 👍